examples/asr/speech_to_text_finetune.py

```diff
@@ -0,0 +1,150 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This script can used to fine-tune a speech-to-text model of any instance type when users want to 
+fine-tune an existing model without changing its core architecture but may change the tokenizer. 
+One can mention the pretrained model in two ways: 
+1) `init_from_nemo_model` or 
+2) `init_from_pretrained_model` in the configuration.
+
+To update the model architecture in conjunction with other modifications, it is advisable to use the primary 'speech_to_text_rnnt/ctc_*.py' script.
+
+Note: To create a single script for all model types, we currently only support two types of 
+initializations:
+1) `init_from_nemo_model`, and
+2) `init_from_pretrained_model`,
+but not `init_from_ptl_ckpt`.
+
+To train with prior base model tokenizer keep `model.tokenizer.update_tokenizer` as false else
+make it true and provide tokenizer dir along with tokenizer type.
+
+To fine-tune the model, use the following commands:
+
+For initialization from a NEMO model:
+```sh
+python <NEMO_ROOT>/examples/asr/speech_to_text_finetune.py \
+    init_from_nemo_model=<path_to_nemo_model>
+```
+
+For initialization from a pretrained model:
+```sh
+python <NEMO_ROOT>/examples/asr/speech_to_text_finetune.py \
+    init_from_pretrained_model=<pretrained_model_name>
+```
+
+# Fine-Tune a Model
+
+For documentation on fine-tuning this model, please visit:
+https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/main/asr/configs.html#fine-tuning-configurations
+"""
+
+import pytorch_lightning as pl
+from omegaconf import OmegaConf
+from pytorch_lightning.utilities import rank_zero_only
+
+from nemo.collections.asr.models import ASRModel
+from nemo.core.config import hydra_runner
+from nemo.utils import logging, model_utils
+from nemo.utils.exp_manager import exp_manager
+
+
+@rank_zero_only
+def get_base_model(cfg):
+    asr_model = None
+    nemo_model_path = cfg.get('init_from_nemo_model', None)
+    pretrained_name = cfg.get('init_from_pretrained_model', None)
+    if nemo_model_path is not None and pretrained_name is not None:
+        raise ValueError("Only pass `init_from_nemo_model` or `init_from_pretrained_model` but not both")
+    elif nemo_model_path is None and pretrained_name is None:
+        raise ValueError(
+            "Both `init_from_nemo_model` and `init_from_pretrained_model cannot be None, should pass atleast one of them"
+        )
+    elif nemo_model_path is not None:
+        asr_model = ASRModel.restore_from(restore_path=nemo_model_path)
+    elif pretrained_name is not None:
+        asr_model = ASRModel.from_pretrained(model_name=pretrained_name)
+
+    return asr_model
+
+
+def update_tokenizer(asr_model, tokenizer_dir, tokenizer_type):
+    vocab_size = asr_model.tokenizer.vocab_size
```

Comment:
Can we make no assumption about Char vs Subword models ? Lots of people seem to use char models